packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/template/StreamSerializationModelTemplate.java

```diff
@@ -726,15 +726,8 @@ private static void serializeJsonProperty(JavaBlock methodBlock, ClientModelProp
                     methodBlock.line("JsonMergePatchHelper.get" + clientType.toString()
                         + "Accessor().prepareModelForJsonMergePatch(" + propertyValueGetter + ", true);");
                 }
-                if (fromSuperType && clientType != wireType && clientType.isNullable()) {
-                    // If the property is from a super type and the client type is different from the wire type then a
-                    // null
-                    // check is required to prevent a NullPointerException when converting the value.
-                    methodBlock.ifBlock(property.getGetterName() + "() != null",
-                        ifAction -> ifAction.line(fieldSerializationMethod + ";"));
-                } else {
-                    methodBlock.line(fieldSerializationMethod + ";");
-                }
+                writeSerializeJsonPropertyViaFieldSerializationMethod(methodBlock, property, model, serializedName,
```

Comment:
No logic change to this file.

Change in this file is to extract this `writeSerializeJsonPropertyViaFieldSerializationMethod` method, for overload in mgmt code path.